x-pack/legacy/plugins/siem/server/types.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+
+export interface ServerFacade {
+  config: Legacy.Server['config'];
+  getInjectedUiAppVars: Legacy.Server['getInjectedUiAppVars'];
+  indexPatternsServiceFactory: Legacy.Server['indexPatternsServiceFactory'];
+  injectUiAppVars: Legacy.Server['injectUiAppVars'];
+  newPlatform: {
```

Comment:
even though you access `newPlatform` from the legacy server object, this is just a work-around until your plugin is moved over. So I would pull this off the legacy server, but not include it in your `ServerFacade` type, that way this type only contains legacy dependencies that you still need to migrate over to NP Core API's.

Reply:
b9bf5ba